experiments/Tammy/models/GRU_Tammy.py

```diff
@@ -0,0 +1,189 @@
+import torch
+import torch.nn as nn
+import torch.nn.init
+
+
+class GRUCell(nn.Module):
+    """
+    Single cell of GRU
+    """
+
+    def __init__(self, input_size: int, hidden_size: int, bias: bool = True):
+        """
+        Initialize gated recurrent unit cell
+        Parameters
+        --------
+          input_size: int
+            The number of expected features in the input x
+          hidden_size: int
+            The number of features in the hidden state h
+          bias: bool
+            Optional, if False,the layer doesn't use bias weights b_ih and b_hh
+            Default: True
+        Returns
+        -------
+            None
+
+        """
+        super(GRUCell, self).__init__()
+        self.input_size = input_size
+        self.hidden_size = hidden_size
+        self.bias = bias
+
+        # reset gate (r)
+        self.reset_i2r = nn.Linear(input_size, hidden_size, bias=bias)
+        self.reset_h2r = nn.Linear(hidden_size, hidden_size, bias=bias)
+
+        # update gate (z)
+        self.update_i2z = nn.Linear(input_size, hidden_size, bias=bias)
+        self.update_h2z = nn.Linear(hidden_size, hidden_size, bias=bias)
+
+        # almost output (n)
+        self.output_i2n = nn.Linear(input_size, hidden_size, bias=bias)
+        self.output_h2n = nn.Linear(hidden_size, hidden_size, bias=bias)
```

Comment:
Biases should be declared independently instead of nesting in these variables.
For example, in ```r= acti(x_t + hs_pre)```, the formula you _actually_ creating is ```tanh(W_i2r*x + b_1 + W_h2r*h + b_2)```